sklearn/preprocessing/_discretization.py

```diff
@@ -341,22 +285,21 @@ def inverse_transform(self, Xt):
         """
         check_is_fitted(self, ["bin_edges_"])
 
-        # Currently, OneHotEncoder doesn't support inverse_transform
         if self.encode != 'ordinal':
             raise ValueError("inverse_transform only supports "
                              "'encode = ordinal'. Got encode={!r} instead."
                              .format(self.encode))
 
-        Xt = self._validate_X_post_fit(Xt)
-        trans = self.transformed_features_
+        Xt = check_array(Xt, dtype='numeric')
```

Comment:
We should probably use `FLOAT_DTYPES` here, since we modify `Xt` inplace and we may want to put float in it.